Add unit tests for wrapper handlers in tools/serve

qiuzhong opened the pull request:
Add unit tests for all kinds of wrapper handlers in tools/serve/serve.py.
See #12452

qiuzhong committed:
Add unit tests for wrapper handlers in tools/serve

Add unit tests for all kinds of wrapper handlers in tools/serve/serve.py.
See #12452

jgraham commented on tools/wptserve/tests/functional/test_handlers.py:
```diff
@@ -8,8 +8,11 @@
 
 wptserve = pytest.importorskip("wptserve")
-from .base import TestUsingServer, TestUsingH2Server, doc_root
+from .base import TestUsingServer, TestUsingH2Server, \
```
Don't use continuation characters, please.

jgraham commented on tools/wptserve/tests/functional/test_handlers.py:
```diff
@@ -388,5 +391,64 @@ def test_requesting_multiple_resources(self):
 
 
+class TestWorkersHandler(TestWrapperHandlerUsingServer):
+
```
No blank line after the class definition.

jgraham reviewed (changes requested):
This is amazing, thanks!

Only a couple of minor style fixes to request.

sideshowbarker commented:
Travis unittest failures:

* https://travis-ci.org/web-platform-tests/wpt/jobs/452825483
* https://travis-ci.org/web-platform-tests/wpt/jobs/452825484

```
_________ ERROR collecting wptserve/tests/functional/test_handlers.py __________
ImportError while importing test module '/home/travis/build/web-platform-tests/wpt/tools/wptserve/tests/functional/test_handlers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
.tox/py27/lib/python2.7/site-packages/six.py:709: in exec_
    exec("""exec _code_ in _globs_, _locs_""")
wptserve/tests/functional/test_handlers.py:14: in <module>
    from serve.serve import WorkersHandler, WindowHandler, AnyHtmlHandler,\
E   ImportError: No module named serve
```
and
```
ERROR: InvocationError for command '/home/travis/build/web-platform-tests/wpt/tools/.tox/py36/bin/pytest --cov' (exited with code 2)
py36-flake8 create: /home/travis/build/web-platform-tests/wpt/tools/.tox/py36-flake8
py36-flake8 installdeps: -rrequirements_flake8.txt
py36-flake8 installed: flake8==3.6.0,flake8-polyfill==1.0.2,mccabe==0.6.1,pep8-naming==0.7.0,pycodestyle==2.4.0,pyflakes==2.0.0
py36-flake8 run-test-pre: PYTHONHASHSEED='3166492845'
py36-flake8 runtests: commands[0] | flake8 --append-config=py36-flake8.ini
___________________________________ summary ____________________________________
ERROR:   py36: commands failed
```

qiuzhong committed:
Update the unit tests

* Fix the import error
* Improve the code style

sideshowbarker commented:
See https://travis-ci.org/web-platform-tests/wpt/jobs/453768016; unittests still failing for w/ Python3

qiuzhong committed:
Read and write files in binary mode

Read and write files in binary mode so the unit tests can pass in
Python3 environment.

qiuzhong commented:
@jgraham , thanks for your comments. I improved the code style.

@sideshowbarker , the import error and other errors have been fixed.

sideshowbarker commented:
Yeah, c4401e444780e26141d44a7de90b30f2683e13fd is all green now, so when the review is approved, this should be ready to merge to master
